src/examples/zkapps/dex/dex.ts

```diff
@@ -338,14 +267,12 @@ class TokenContract extends SmartContract {
   }
 
   // let a zkapp do whatever it wants, as long as the token supply stays constant
-  @method authorize(callback: Experimental.Callback<any>) {
-    let layout = [[3, 0, 0], 0, 0]; // these are 10 child account updates we allow, in a left-biased tree of width 3
-    // TODO: this should also return what the callback returns, and authorize should pass it on!
-    let zkappUpdate = Experimental.accountUpdateFromCallback(
-      this,
-      layout,
-      callback
-    );
+  @method authorize(zkappUpdate: AccountUpdate) {
+    let layout = [[[3, 0, 0], 0, 0]]; // these are 10 child account updates we allow, in a left-biased tree of width 3
```

Comment:
This part of the API feels really low level to me, what are your thoughts on creating an abstracting around `layout` in a later PR?

I'm not sure how exactly this would look like but some of the cases documented are duplicates of each other in terms of functionality, it would be great to have type/types to capture the intent.

For example:
Perhaps we can have a type called `EmptyAccountUpdate`.

```
* []         // an empty accountUpdates list -- EmptyAccountUpdate
* 0          // same as [] -- EmptyAccountUpdate
```
```
 * [0]        // a list of one accountUpdate, which doesn't have children 
 * 1          // same as [0] 
 * 2          // same as [0, 0] 
```
In these cases, we can have `ChildlessAccountUpdate` and encapsulate all the cases by wrapping them with indexes in a list, e.g. 

 * [0]        `[ChildlessAccountUpdate]`
 * 1          // same as [0] - `[ChildlessAccountUpdate]`
 * 2          // same as [0, 0] - `[ChildlessAccountUpdate, ChildlessAccountUpdate]`
 
 Just to clarify, I'm not suggesting we do this in this PR, just wondering your thoughts on a follow-up later on when we have a better idea of what types/API we want for this?

Reply:
That's a great suggestion, having names instead of numbers will go a long way towards making it clearer :D

Maybe two functions are enough:
* `StaticChildren(n: number)` / `StaticChildren(...children)` to replace 0, 1, .., [0, ...], ...
* `AnyChildren` to replace `null`

Any maybe `NoChildren` as alias for `StaticChildren(0)`

So the layout in the DEX example `[[3, 0, 0], 0, 0]` would become
```ts
// in a namespace to not bloat exports
let { StaticChildren, NoChildren } = AccountUpdateLayout;

let layout = StaticChildren(
  StaticChildren(StaticChildren(3), NoChildren, NoChildren),
  NoChildren,
  NoChildren
);
```

Reply:
or should we be even more verbose (and potentially, clearer) with something like `AccountUpdateWithChildren(3)`, `ChildlessAccountUpdate`, `AccountUpdateWithAnyChildren`?

Reply:
ended up with the shorter names above, but put them under a nested namespace: `AccountUpdate.Layout`